### DISCUSSION TITLE
Intermittent ‘Read-only Database’ Errors with LiteFS in Multi-region

### DISCUSSION THREAD

**Discussion by bryantbrock** [epicweb-dev/epic-stack#753] [Q&A]
I know this is a fly.io question (see my community question [here](https://community.fly.io/t/intermittent-read-only-database-errors-with-litefs-in-multi-region-epic-stack-remix-app/20135)) but maybe the epic-stack community can help if anyone else has run into this issue.

My stack is exactly what's in the epic-stack but, utilizing capacitor.js, is a mobile app currently in TestFlight on iOS.

I've `fly m clone <id>` to scale my app. However, I'm still getting a write error. Every other POST request is erroring with “attempt to write a readonly database”. I’ve scoured the internet, proxy docs ([Built-in HTTP Proxy · Fly Docs](https://fly.io/docs/litefs/proxy/)), and still can’t understand what’s wrong with my setup. Again, every other request succeeds, but the others throw this error. Logs show the same POST request failing the first time and then succeeding the second time. What’s going on?

```
ord [info] level=INFO msg="fuse: create(): cannot create journal: read only replica"
ord [info] prisma:error
ord [info] Invalid `prisma.group.update()` invocation:
ord [info] Error occurred during query execution:
ord [info] ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(SqliteError { extended_code: 1544, message: Some("attempt to write a readonly database") }), transient: false })
ord [info] POST /app/groups/clwpwocse0001yvdd23s9cz5i?_data=routes/app+/groups+/$id/route 500 - - 12.445 ms
ord [info] GET /app/groups/clwpwocse0001yvdd23s9cz5i?_data=root 200 - - 10.639 ms
ord [info] HEAD / 200 - - 6.461 ms
ord [info] HEAD / 200 - - 6.171 ms
ord [info] GET / 302 - - 7.950 ms
ord [info] GET /app 200 - - 49.616 ms
ord [info] GET /app/groups?_data=routes/app+/groups+/_index 200 - - 9.115 ms
ord [info] GET /app/groups/clwpwocse0001yvdd23s9cz5i?_data=routes/app+/groups+/$id/route 200 - - 32.090 ms
ord [info] POST /app/groups/clwpwocse0001yvdd23s9cz5i?_data=routes/app+/groups+/$id/route 204 - - 29.891 ms
ord [info] GET /app/groups?_data=routes/app+/route 200 - - 2.532 ms
ord [info] GET /app/groups?_data=root 200 - - 21.985 ms
ord [info] GET /app/groups?_data=routes/app+/groups+/_index 200 - - 17.755 ms
```

**Comment by kentcdodds** [comment]
Looks like you're figured out the issue! https://github.com/superfly/litefs/issues/386#issuecomment-1685857728
